server/server.go

```diff
@@ -333,7 +336,17 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		return nil, errors.Wrap(err, "initializing webhooks")
 	}
 	vcsClient := vcs.NewClientProxy(githubClient, gitlabClient, bitbucketCloudClient, bitbucketServerClient, azuredevopsClient)
-	commitStatusUpdater := &command.VCSStatusUpdater{Client: vcsClient, TitleBuilder: vcs.StatusTitleBuilder{TitlePrefix: userConfig.VCSStatusName}}
+
+	var commitStatusUpdater events.CommitStatusUpdater
+	commitStatusUpdater = &command.VCSStatusUpdater{Client: vcsClient, TitleBuilder: vcs.StatusTitleBuilder{TitlePrefix: userConfig.VCSStatusName}}
+	if ChecksEnabled {
+		cm := commitStatusUpdater.(*command.VCSStatusUpdater)
+		commitStatusUpdater = &command.ChecksEnabledVCSStatusUpdater{
+			VCSStatusUpdater: *cm,
+			Client:           vcsClient,
+			TitleBuilder:     vcs.StatusTitleBuilder{TitlePrefix: userConfig.VCSStatusName},
```

Comment:
so after doing some thinking I realized this has much bigger implications than you might realize. Up until now, Atlantis has implicitly been able to handle supporting multiple providers at once, (ie. receiving webhooks from both github, gitlab on the same instance).  This changes that assumption since you're injecting an implementation based on the provider.

I think instead it might be easier to just have this logic at the client level like i mentioned in the last PR.